### ISSUE TITLE
Failed to find any class that implements Connector and which name matches io.confluent.connect.mqtt.MqttSourceConnector

### ISSUE THREAD

**Issue by raghubanda** [confluentinc/cp-docker-images#745]
Hi @arrawatia, have you seen this by any chance?
curl -X POST \
  http://localhost:8083/connectors \
  -H 'Accept: */*' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Host: localhost:8083' \
  -H 'User-Agent: PostmanRuntime/7.15.0' \
  -H 'accept-encoding: gzip, deflate' \
  -H 'content-length: 403' \
  -H 'cookie: XSRF-TOKEN=70197154-2a98-4f1b-8701-50c49b6be804; JSESSIONID=C6B2F72F927A21DD6C97A3318F376B9B' \
  -b 'XSRF-TOKEN=70197154-2a98-4f1b-8701-50c49b6be804; JSESSIONID=C6B2F72F927A21DD6C97A3318F376B9B' \
  -d '{
    "name" : "mqtt-source",
"config" : {
    "connector.class" : "io.confluent.connect.mqtt.MqttSourceConnector",
    "tasks.max" : "1",
    "mqtt.server.uri" : "tcp://127.0.0.1:1883",
    "mqtt.topics" : "temperature",
    "kafka.topic" : "mqtt.temperature",
    "confluent.topic.bootstrap.servers": "localhost:9092",
    "confluent.topic.replication.factor": "1",
    "confluent.license":""
    }
}'\'''

Response
{
    "error_code": 500,
    "message": "Failed to find any class that implements Connector and which name matches io.confluent.connect.mqtt.MqttSourceConnector, available connectors are: PluginDesc{klass=class io.confluent.connect.activemq.ActiveMQSourceConnector, name='io.confluent.connect.activemq.ActiveMQSourceConnector', version='5.2.1', encodedVersion=5.2.1, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-activemq/'}, PluginDesc{klass=class io.confluent.connect.elasticsearch.ElasticsearchSinkConnector, name='io.confluent.connect.elasticsearch.ElasticsearchSinkConnector', version='5.2.1', encodedVersion=5.2.1, type=sink, typeName='sink', location='file:/usr/share/java/kafka-connect-elasticsearch/'}, PluginDesc{klass=class io.confluent.connect.gcs.GcsSinkConnector, name='io.confluent.connect.gcs.GcsSinkConnector', version='5.0.1', encodedVersion=5.0.1, type=sink, typeName='sink', location='file:/usr/share/confluent-hub-components/confluentinc-kafka-connect-gcs/'}, PluginDesc{klass=class io.confluent.connect.hdfs.HdfsSinkConnector, name='io.confluent.connect.hdfs.HdfsSinkConnector', version='5.2.1', encodedVersion=5.2.1, type=sink, typeName='sink', location='file:/usr/share/java/kafka-connect-hdfs/'}, PluginDesc{klass=class io.confluent.connect.hdfs.tools.SchemaSourceConnector, name='io.confluent.connect.hdfs.tools.SchemaSourceConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-hdfs/'}, PluginDesc{klass=class io.confluent.connect.ibm.mq.IbmMQSourceConnector, name='io.confluent.connect.ibm.mq.IbmMQSourceConnector', version='5.2.1', encodedVersion=5.2.1, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-ibmmq/'}, PluginDesc{klass=class io.confluent.connect.jdbc.JdbcSinkConnector, name='io.confluent.connect.jdbc.JdbcSinkConnector', version='5.2.1', encodedVersion=5.2.1, type=sink, typeName='sink', location='file:/usr/share/java/kafka-connect-jdbc/'}, PluginDesc{klass=class io.confluent.connect.jdbc.JdbcSourceConnector, name='io.confluent.connect.jdbc.JdbcSourceConnector', version='5.2.1', encodedVersion=5.2.1, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-jdbc/'}, PluginDesc{klass=class io.confluent.connect.jms.JmsSourceConnector, name='io.confluent.connect.jms.JmsSourceConnector', version='5.2.1', encodedVersion=5.2.1, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-activemq/'}, PluginDesc{klass=class io.confluent.connect.s3.S3SinkConnector, name='io.confluent.connect.s3.S3SinkConnector', version='5.2.1', encodedVersion=5.2.1, type=sink, typeName='sink', location='file:/usr/share/java/kafka-connect-s3/'}, PluginDesc{klass=class io.confluent.connect.storage.tools.SchemaSourceConnector, name='io.confluent.connect.storage.tools.SchemaSourceConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-s3/'}, PluginDesc{klass=class io.confluent.kafka.connect.datagen.DatagenConnector, name='io.confluent.kafka.connect.datagen.DatagenConnector', version='null', encodedVersion=null, type=source, typeName='source', location='file:/usr/share/confluent-hub-components/confluentinc-kafka-connect-datagen/'}, PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSinkConnector, name='org.apache.kafka.connect.file.FileStreamSinkConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=sink, typeName='sink', location='file:/usr/share/java/kafka/'}, PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSourceConnector, name='org.apache.kafka.connect.file.FileStreamSourceConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=source, typeName='source', location='file:/usr/share/java/kafka/'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockConnector, name='org.apache.kafka.connect.tools.MockConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=connector, typeName='connector', location='file:/usr/share/java/kafka-connect-elasticsearch/'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSinkConnector, name='org.apache.kafka.connect.tools.MockSinkConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=sink, typeName='sink', location='file:/usr/share/java/kafka-connect-elasticsearch/'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSourceConnector, name='org.apache.kafka.connect.tools.MockSourceConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-elasticsearch/'}, PluginDesc{klass=class org.apache.kafka.connect.tools.SchemaSourceConnector, name='org.apache.kafka.connect.tools.SchemaSourceConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-elasticsearch/'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSinkConnector, name='org.apache.kafka.connect.tools.VerifiableSinkConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-elasticsearch/'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSourceConnector, name='org.apache.kafka.connect.tools.VerifiableSourceConnector', version='2.2.0-cp2', encodedVersion=2.2.0-cp2, type=source, typeName='source', location='file:/usr/share/java/kafka-connect-elasticsearch/'}"
}

**Comment by cricket007** [comment]
Could you please clarify how you installed the MQTT source connector into your Docker image or why you thought it was already available if you didn't install it?

And if it did work, then `mqtt.server.uri" : "tcp://127.0.0.1:1883` is incorrect, and you'd need to use the address for the actual MQTT broker, not the container itself

**Comment by lbchenthu** [comment]
I encountered the same problem. Somebody can help? 

I used docker-compose to install confluent and 
used  confluent-hub to install MQTT plugin (confluentinc/kafka-connect-mqtt:1.2.3). 

The mqtt connector does not show in the "Connector" page of control center

**Comment by cricket007** [comment]
@lbchenthu Please show your docker-compose or dockerfile as part of a [SSCCE](https://sscce.org)

**Comment by lbchenthu** [comment]
I download the file from "Quick Start" page,  
https://docs.confluent.io/current/quickstart/ce-docker-quickstart.html
I used the "cp-all-in-one" file:  
https://github.com/confluentinc/examples/tree/5.4.1-post/cp-all-in-one

**Comment by cricket007** [comment]
Yes, but that doesn't come with the MQTT Connector, to my knowledge. Please show how you installed it

**Comment by lbchenthu** [comment]
Thank you for your effort.
After restarting the "connect" worker, the mqtt-connector shows on the dashboard correctly.

The steps are as follows.
1. log into "connect" worker and type command 
"confluent-hub install confluentinc/kafka-connect-mqtt:1.2.3" 

2. The "plugin.path" property is set to "/usr/share/confluent-hub-components"

MYHOST:~/examples/cp-all-in-one$ sudo docker-compose exec connect "/bin/bash"
root@connect:/# cd /usr/share/confluent-hub-components/

root@connect:/usr/share/confluent-hub-components# ls

confluentinc-kafka-connect-datagen  confluentinc-kafka-connect-gcs  confluentinc-kafka-connect-mqtt


root@connect:/usr/share/confluent-hub-components# cat /etc/kafka-connect/kafka-connect.properties
....
plugin.path=/usr/share/java,/usr/share/confluent-hub-components
...

**Comment by cricket007** [comment]
Yeah, that won't work. You have to restart Connect process for any plugin to be discovered. 

You would have to write your own Dockerfile. See https://github.com/confluentinc/cp-docker-images/blob/5.3.1-post/debian/kafka-connect/Dockerfile#L41 for example
